com.unity.ml-agents.extensions/Runtime/Match3/Match3Sensor.cs

```diff
@@ -113,73 +137,58 @@ public ObservationSpec GetObservationSpec()
         /// <inheritdoc/>
         public int Write(ObservationWriter writer)
         {
-            // if (m_Board.Rows != m_Rows || m_Board.Columns != m_Columns || m_Board.NumCellTypes != m_NumCellTypes)
-            // {
-            //     Debug.LogWarning(
-            //         $"Board shape changes since sensor initialization. This may cause unexpected results. " +
-            //         $"Old shape: Rows={m_Rows} Columns={m_Columns}, NumCellTypes={m_NumCellTypes} " +
-            //         $"Current shape: Rows={m_Board.Rows} Columns={m_Board.Columns}, NumCellTypes={m_Board.NumCellTypes}"
-            //     );
-            // }
-
-            if (m_ObservationType == Match3ObservationType.Vector)
+            CheckBoardSizes();
+            var currentBoardSize = m_Board.GetCurrentBoardSize();
+
+            int offset = 0;
+            var isVisual = m_ObservationType != Match3ObservationType.Vector;
+            for (var r = 0; r < m_MaxBoardSize.Rows; r++)
             {
-                int offset = 0;
-                for (var r = 0; r < m_Rows; r++)
+                for (var c = 0; c < m_MaxBoardSize.Columns; c++)
                 {
-                    for (var c = 0; c < m_Columns; c++)
+                    if (r < currentBoardSize.Rows && c < currentBoardSize.Columns)
```

Comment:
not a requirement at all, but could you also eliminate this branch by multiplying the value by something like
```
val = val * (Math.Min(Math.Max(r - currentBoardSize.Rows, 0), 1));
val = val * (Math.Min(Math.Max(c - currentBoardSize.Columns, 0), 1));
write.WriteOneHot(offset, r, c, val, m_oneHotSize, isVisual);
```

Reply:
(discussed offline - I think this particular change is incorrect).